applications/rovioli/src/rovio-localization-handler.cc

```diff
@@ -246,13 +261,65 @@ bool RovioLocalizationHandler::processAsUpdate(
     measurement_accepted = rovio_interface_->processGroundTruthUpdate(
         JrJV, qJV, rovio_timestamp_sec);
   } else {
+    // Check if there are any matches to be processed in the camera frames that
+    // are used by ROVIO for estimation (inactive).
+    const size_t num_cameras =
+        localization_result->G_landmarks_per_camera.size();
+    size_t num_valid_matches = 0u;
+    for (size_t cam_idx = 0u; cam_idx < num_cameras; ++cam_idx) {
+      const size_t* rovio_cam_idx =
+          maplab_to_rovio_cam_indices_mapping_.getRight(cam_idx);
+      if (rovio_cam_idx == nullptr) {
+        continue;
+      }
+      num_valid_matches +=
+          localization_result->G_landmarks_per_camera[cam_idx].cols();
+    }
+    if (num_valid_matches == 0u) {
+      // There are no valid localization matches for the cameras used by ROVIO.
+      // Use the localization result of the inactive cameras but integrate them
+      // using the 6dof localization mode, since we cannot currently pass 2D-3D
+      // correspondences to ROVIO for a camera it's not using.
```

Comment:
Maybe we can just initialise the camera but not pass images but still 2d-3d constraints? I'll take a look if that is possible later.

Reply:
I agree, this is clearly the better solution, but let's keep this for now, it's anyways optional.